R/getPlot.R

```diff
@@ -0,0 +1,121 @@
+#'   display.
+#' - gateLabel character One of "eventcount", "mean", "median", "percent",
+#'   "mad", "cv", "stddev", "geometricmean", "name", "none".
+#' - gateLabelFontSize double Font size for gate labels.
+#' - height double Image height. Defaults to 228.
+#' - percentileStart double For contour plots, the percentile of the first
+#'   contour.
+#' - percentileStep double For contour plots, the percentile step between
+#'   each contour.
+#' - populationId character Population ID. Defaults to ungated.
+#' - postSubsampleN double Randomly subsample the file to contain this many
+#'   events after gating.
+#' - postSubsampleP double Randomly subsample the file to contain this
+#'   percent of events (0 to 1) after gating.
+#' - renderGates logical Render gates into the image.
+#' - seed double Seed for random number generator used for subsampling. Use
+#'   for deterministic reproducible subsampling. If omitted, a pseudo-random
+#'   value is used.
+#' - smoothing double For density and contour plots, adjusts the amount of
+#'   smoothing. Defaults to 0 (no smoothing). Set to 1 for typical smoothing.
+#'   Higher values (up to 10) increase smoothing.
+#' - strokeThickness double The thickness of histogram and contour plot
+#'   lines. Defaults to 1.
+#' - tickLabelsQ logical Display tick labels. Defaults to false.
+#' - ticksQ logical Display ticks. Defaults to true.
+#' - width Integer Image width. Defaults to 228.
+#' - xAxisLabelQ logical Display x axis label. Overrides axisLabelsQ.
+#' - xAxisQ logical Display x axis line. Overrides axesQ.
+#' - xTickLabelsQ logical Display x tick labels. Overrides tickLabelsQ.
+#' - xTicksQ logical Display x ticks. Overrides ticksQ.
+#' - yAxisLabel logical Display y axis label. Overrides axisLabelsQ.
+#' - yAxisQ logical Display y axis line. Overrides axesQ.
+#' - yTickLabelsQ logical Display y tick labels. Overrides tickLabelsQ.
+#' - yTicksQ logical Display y ticks. Overrides ticksQ.
+#' @param destination character If specified, write the image to file, rather than
+#'   display it.
+#' @param overwrite logical If a destination is specified, allows destination
+#'   file to be overwritten.
+#' @param raw logical Return the image as raw bytes.
+#'
+#' @md
+#' @export
+#' @examples
+#' \dontrun{
+#' parameters <- list(
+#'   "width" = 400,
+#'   "height" = 400,
+#'   "axesQ" = False,
+#'   "ticksQ" = False,
+#'   "color" = "#ff0000",
+#'   "renderGates" = True,
+#'   "preSubsampleN" = 4,
+#'   "preSubsampleP" = 0.5,
+#'   "postSubsampleN" = 4,
+#'   "postSubsampleP" = 0.5,
+#'   "seed" = 9,
+#'   "smoothing" = 0.75,
+#' )
+#' getPlot(experimentId, fcsFileId, plotType = "dot", xChannel = "FSC-A", yChannel = "FSC-W", params = parameters)
+#' }
+getPlot <- function(experimentId,
+                    fcsFileId,
+                    plotType,
+                    xChannel,
+                    yChannel = "",
+                    zChannel = "",
+                    params = list(),
```

Comment:
I don't recall why our API defaults to no axis labels and no ticks. Can you merge these in to the user's params to match our UI defaults? `axisLabelsQ=T`, `ticksQ=T`

Reply:
When I pass these directly, I still don't get labels and ticks. It seems that the API should actually default to showing labels and ticks, too. Troubleshooting...

Reply:
Probably R logical -> "true"/"false" string conversion needed.

Reply:
Okay, figured it out:
```r
> res$request$url
[1] "https://cellengine.com/api/v1/experiments/5d2f8b4b21fd0676fb3a6a70/plot?fcsFileId=5d2f8b4b21fd0676fb3a6a72&plotType=dot&xChannel=FSC-A&yChannel=FSC-W&zChannel=&ticksQ=TRUE"
```

Reply:
n0te "ticksQ=TRUE"